### ISSUE TITLE
Fact gathering fails due to path in /proc/mounts containing quote

### ISSUE THREAD

**Issue by ariep** [ansible/ansible#16855]
##### ISSUE TYPE
- Bug Report
##### ANSIBLE VERSION

```
ansible 2.1.0.0
```
##### OS / ENVIRONMENT

Both controller and remote are Gentoo Linux; I expect that for the controller this is unimportant, and for the remote any type of linux will have the same problem.
##### SUMMARY

When the remote has a mounted file system with the device path containing a single quote `'`, fact gathering fails.
##### STEPS TO REPRODUCE
1. On a host managed by ansible, create and mount a file system with device path containing a single quote, for example by using sshfs.
2. Verify that `/proc/mounts` on the remote contains a line with a quote `'`.
3. From the controller, do

```
ansible remoteHost -m setup
```

, or run any playbook that includes this remote as host.
##### EXPECTED RESULTS

Ansible outputs the usual list of facts from the remote host.
##### ACTUAL RESULTS

Ansible fails with the following error:

```
Loaded callback minimal of type stdout, v2.0
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: arie
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1469708255.88-95535704478326 `" && echo ansible-tmp-1469708255.88-95535704478326="` echo $HOME/.ansible/tmp/ansible-tmp-1469708255.88-95535704478326 `" ) && sleep 0'                                                                                                        
<127.0.0.1> PUT /tmp/tmpEAP3Xv TO /home/arie/.ansible/tmp/ansible-tmp-1469708255.88-95535704478326/setup
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_GB.utf8 LC_ALL=en_GB.utf8 LC_MESSAGES=en_GB.utf8 /usr/bin/python2.7 /home/arie/.ansible/tmp/ansible-tmp-1469708255.88-95535704478326/setup; rm -rf "/home/arie/.ansible/tmp/ansible-tmp-1469708255.88-95535704478326/" > /dev/null 2>&1 && sleep 0'                                                                                
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):                                                                                                                                                  
  File "/tmp/ansible_TVUXtC/ansible_module_setup.py", line 127, in <module>                                                                                                         
    main()                                                                                                                                                                          
  File "/tmp/ansible_TVUXtC/ansible_module_setup.py", line 119, in main                                                                                                             
    data = get_all_facts(module)                                                                                                                                                    
  File "/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py", line 3177, in get_all_facts                                                                          
  File "/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py", line 3123, in ansible_facts                                                                          
  File "/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py", line 919, in populate                                                                                
  File "/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py", line 107, in wrapper                                                                                 
  File "/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py", line 1151, in get_mount_facts                                                                        
  File "/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/basic.py", line 2034, in run_command                                                                            
  File "/usr/lib64/python2.7/shlex.py", line 279, in split                                                                                                                          
    return list(lex)                                                                                                                                                                
  File "/usr/lib64/python2.7/shlex.py", line 269, in next                                                                                                                           
    token = self.get_token()                                                                                                                                                        
  File "/usr/lib64/python2.7/shlex.py", line 96, in get_token                                                                                                                       
    raw = self.read_token()                                                                                                                                                         
  File "/usr/lib64/python2.7/shlex.py", line 172, in read_token                                                                                                                     
    raise ValueError, "No closing quotation"                                                                                                                                        
ValueError: No closing quotation                                                                                                                                                    

localhost | FAILED! => {
    "changed": false,                                                                                                                                                               
    "failed": true,                                                                                                                                                                 
    "invocation": {                                                                                                                                                                 
        "module_name": "setup"                                                                                                                                                      
    },                                                                                                                                                                              
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_TVUXtC/ansible_module_setup.py\", line 127, in <module>\n    main()\n  File \"/tmp/ansible_TVUXtC/ansible_module_setup.py\", line 119, in main\n    data = get_all_facts(module)\n  File \"/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py\", line 3177, in get_all_facts\n  File \"/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py\", line 3123, in ansible_facts\n  File \"/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py\", line 919, in populate\n  File \"/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py\", line 107, in wrapper\n  File \"/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/facts.py\", line 1151, in get_mount_facts\n  File \"/tmp/ansible_TVUXtC/ansible_modlib.zip/ansible/module_utils/basic.py\", line 2034, in run_command\n  File \"/usr/lib64/python2.7/shlex.py\", line 279, in split\n    return list(lex)\n  File \"/usr/lib64/python2.7/shlex.py\", line 269, in next\n    token = self.get_token()\n  File \"/usr/lib64/python2.7/shlex.py\", line 96, in get_token\n    raw = self.read_token()\n  File \"/usr/lib64/python2.7/shlex.py\", line 172, in read_token\n    raise ValueError, \"No closing quotation\"\nValueError: No closing quotation\n",                                                                                                                   
    "module_stdout": "",                                                                                                                                                            
    "msg": "MODULE FAILURE",                                                                                                                                                        
    "parsed": false                                                                                                                                                                 
}
```

**Comment by jctanner** [comment]
@ariep Can you show us your example /proc/mounts content?

**Comment by ariep** [comment]
This is the line from `/proc/mounts` that causes trouble:

```
selfukpi:/mnt/data/foto's /mnt/data/home/user/MOUNTNAME fuse.sshfs rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
```

After unmounting this particular mount, ansible works again.

**Comment by ariep** [comment]
Not sure why this still has the needs_info label. Anything more I can do to help?

**Comment by alikins** [comment]
Tested with the following mount points:

```
grimlock.g.a:test_path/path_with'single_quotes /home/adrian/sshfs-grimlock-single-quote fuse.sshfs rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
grimlock.g.a:path_with'single_quotes /home/adrian/sshfs-grimlock-single-quote-2 fuse.sshfs rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
```

Testing with the devel branch, the stable-2.1 branch, and the 2.1.0.1-1 branch 
a 'ansible localhost -m setup' runs without errors. The sshfs mounts aren't included in the mount facts, but that is intentional.

But, if I add:

```
grimlock.g.a:/mnt/data/foto's /home/adrian/fotos fuse.sshfs rw,nosuid,nodev,relatime,user_id=1000,group_id=1000 0 0
```

I can get this to fail on 2.1.0.0-1, but stable-2.1 and devel branches work.

```
[fedora-23:ansible-2.1.0.0-1 (v2.1.0.0-1-branch)]$ ansible me -vvvvv -i ~/src/ansible/hosts  -m setup
Using /home/adrian/ansible/ansible.cfg as config file
Loaded callback minimal of type stdout, v2.0
<127.0.0.1> ESTABLISH LOCAL CONNECTION FOR USER: adrian
<127.0.0.1> EXEC /bin/sh -c '( umask 77 && mkdir -p "` echo $HOME/.ansible/tmp/ansible-tmp-1474987242.49-127237382178307 `" && echo ansible-tmp-1474987242.49-127237382178307="` echo $HOME/.ansible/tmp/ansible-tmp-1474987242.49-127237382178307 `" ) && sleep 0'
<127.0.0.1> PUT /tmp/tmpPJggpL TO /home/adrian/.ansible/tmp/ansible-tmp-1474987242.49-127237382178307/setup
<127.0.0.1> EXEC /bin/sh -c 'LANG=en_US.utf8 LC_ALL=en_US.utf8 LC_MESSAGES=en_US.utf8 /usr/bin/python /home/adrian/.ansible/tmp/ansible-tmp-1474987242.49-127237382178307/setup; rm -rf "/home/adrian/.ansible/tmp/ansible-tmp-1474987242.49-127237382178307/" > /dev/null 2>&1 && sleep 0'
An exception occurred during task execution. The full traceback is:
Traceback (most recent call last):
  File "/tmp/ansible_AFJd_7/ansible_module_setup.py", line 127, in <module>
    main()
  File "/tmp/ansible_AFJd_7/ansible_module_setup.py", line 119, in main
    data = get_all_facts(module)
  File "/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py", line 3177, in get_all_facts
  File "/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py", line 3123, in ansible_facts
  File "/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py", line 919, in populate
  File "/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py", line 107, in wrapper
  File "/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py", line 1151, in get_mount_facts
  File "/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/basic.py", line 2034, in run_command
  File "/usr/lib64/python2.7/shlex.py", line 279, in split
    return list(lex)
  File "/usr/lib64/python2.7/shlex.py", line 269, in next
    token = self.get_token()
  File "/usr/lib64/python2.7/shlex.py", line 96, in get_token
    raw = self.read_token()
  File "/usr/lib64/python2.7/shlex.py", line 172, in read_token
    raise ValueError, "No closing quotation"
ValueError: No closing quotation

me | FAILED! => {
    "changed": false, 
    "failed": true, 
    "invocation": {
        "module_name": "setup"
    }, 
    "module_stderr": "Traceback (most recent call last):\n  File \"/tmp/ansible_AFJd_7/ansible_module_setup.py\", line 127, in <module>\n    main()\n  File \"/tmp/ansible_AFJd_7/ansible_module_setup.py\", line 119, in main\n    data = get_all_facts(module)\n  File \"/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py\", line 3177, in get_all_facts\n  File \"/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py\", line 3123, in ansible_facts\n  File \"/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py\", line 919, in populate\n  File \"/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py\", line 107, in wrapper\n  File \"/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/facts.py\", line 1151, in get_mount_facts\n  File \"/tmp/ansible_AFJd_7/ansible_modlib.zip/ansible/module_utils/basic.py\", line 2034, in run_command\n  File \"/usr/lib64/python2.7/shlex.py\", line 279, in split\n    return list(lex)\n  File \"/usr/lib64/python2.7/shlex.py\", line 269, in next\n    token = self.get_token()\n  File \"/usr/lib64/python2.7/shlex.py\", line 96, in get_token\n    raw = self.read_token()\n  File \"/usr/lib64/python2.7/shlex.py\", line 172, in read_token\n    raise ValueError, \"No closing quotation\"\nValueError: No closing quotation\n", 
    "module_stdout": "", 
    "msg": "MODULE FAILURE", 
    "parsed": false
}
```

**Comment by alikins** [comment]
I'm also going to add the mtab with single quotes to the unit tests to make sure it doesn't break in the future.

**Comment by alikins** [comment]
Since this appears to be fixed in devel branch, and in stable-2.1 (the branch for the next 2.1 release), I'm going to close this. 

You can test the https://github.com/ansible/ansible/releases/tag/v2.1.2.0-0.4.rc4 release candidate to verify. If that doesn't fix it, please reopen this bug.
